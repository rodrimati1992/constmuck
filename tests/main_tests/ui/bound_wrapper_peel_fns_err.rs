use bytemuck::TransparentWrapper;

struct Foo<T: ?Sized>(T);






fn _test_peel<T>(a: Foo<T>, b: Foo<T>) -> (T, T) {
    (
        constmuck::wrapper::peel::<Foo<T>, T>(a),
        <Foo<T> as TransparentWrapper<T>>::peel(b),
    )
}




fn _test_peel_ref<'a, T>(a: &'a Foo<T>, b: &'a Foo<T>, c: &'a Foo<T>) -> (&'a T, &'a T, &'a T) {
    (
        constmuck::wrapper::peel_ref::<Foo<T>, T>(a),
        constmuck::wrapper::peel_ref!(b, Foo<T>, T),
        <Foo<T> as TransparentWrapper<T>>::peel_ref(c),
    )
}



fn _test_peel_slice<'a, T>(a: &'a [Foo<T>], b: &'a [Foo<T>]) -> (&'a [T], &'a [T]) {
    (
        constmuck::wrapper::peel_slice::<Foo<T>, T>(a),
        <Foo<T> as TransparentWrapper<T>>::peel_slice(b),
    )
}




fn main(){}