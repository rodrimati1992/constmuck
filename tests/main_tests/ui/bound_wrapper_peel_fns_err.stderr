error[E0277]: the trait bound `Foo<T>: TransparentWrapper<T>` is not satisfied
   --> tests/main_tests/ui/bound_wrapper_peel_fns_err.rs:12:47
    |
12  |         constmuck::wrapper::peel::<Foo<T>, T>(a),
    |         ------------------------------------- ^ the trait `TransparentWrapper<T>` is not implemented for `Foo<T>`
    |         |
    |         required by a bound introduced by this call
    |
    = help: the trait `TransparentWrapper<T>` is implemented for `Wrapping<T>`
note: required by a bound in `constmuck::wrapper::peel`
   --> src/wrapper.rs:170:12
    |
168 | pub const fn peel<Outer, Inner>(val: Outer) -> Inner
    |              ---- required by a bound in this function
169 | where
170 |     Outer: TransparentWrapper<Inner>,
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `peel`

error[E0277]: the trait bound `Foo<T>: TransparentWrapper<T>` is not satisfied
  --> tests/main_tests/ui/bound_wrapper_peel_fns_err.rs:13:49
   |
13 |         <Foo<T> as TransparentWrapper<T>>::peel(b),
   |         --------------------------------------- ^ the trait `TransparentWrapper<T>` is not implemented for `Foo<T>`
   |         |
   |         required by a bound introduced by this call
   |
   = help: the trait `TransparentWrapper<T>` is implemented for `Wrapping<T>`

error[E0277]: the trait bound `Foo<T>: TransparentWrapper<T>` is not satisfied
   --> tests/main_tests/ui/bound_wrapper_peel_fns_err.rs:22:51
    |
22  |         constmuck::wrapper::peel_ref::<Foo<T>, T>(a),
    |         ----------------------------------------- ^ the trait `TransparentWrapper<T>` is not implemented for `Foo<T>`
    |         |
    |         required by a bound introduced by this call
    |
    = help: the trait `TransparentWrapper<T>` is implemented for `Wrapping<T>`
note: required by a bound in `constmuck::wrapper::peel_ref`
   --> src/wrapper.rs:206:12
    |
204 | pub const fn peel_ref<Outer, Inner>(reff: &Outer) -> &Inner
    |              -------- required by a bound in this function
205 | where
206 |     Outer: TransparentWrapper<Inner>,
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `peel_ref`

error[E0277]: the trait bound `Foo<T>: TransparentWrapper<T>` is not satisfied
  --> tests/main_tests/ui/bound_wrapper_peel_fns_err.rs:23:9
   |
23 |         constmuck::wrapper::peel_ref!(b, Foo<T>, T),
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `TransparentWrapper<T>` is not implemented for `Foo<T>`
   |
   = help: the trait `TransparentWrapper<T>` is implemented for `Wrapping<T>`
note: required by a bound in `constmuck_internal::__PeelRefArgs::<'a, Outer, Inner>::__call`
  --> constmuck_internal/src/lib.rs:74:16
   |
72 |     pub const unsafe fn __call(self) -> &'a Inner
   |                         ------ required by a bound in this associated function
73 |     where
74 |         Outer: TransparentWrapper<Inner>
   |                ^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `__PeelRefArgs::<'a, Outer, Inner>::__call`
   = note: this error originates in the macro `$crate::wrapper_inner` which comes from the expansion of the macro `constmuck::wrapper::peel_ref` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `Foo<T>: TransparentWrapper<T>` is not satisfied
  --> tests/main_tests/ui/bound_wrapper_peel_fns_err.rs:24:53
   |
24 |         <Foo<T> as TransparentWrapper<T>>::peel_ref(c),
   |         ------------------------------------------- ^ the trait `TransparentWrapper<T>` is not implemented for `Foo<T>`
   |         |
   |         required by a bound introduced by this call
   |
   = help: the trait `TransparentWrapper<T>` is implemented for `Wrapping<T>`

error[E0277]: the trait bound `Foo<T>: TransparentWrapper<T>` is not satisfied
   --> tests/main_tests/ui/bound_wrapper_peel_fns_err.rs:32:53
    |
32  |         constmuck::wrapper::peel_slice::<Foo<T>, T>(a),
    |         ------------------------------------------- ^ the trait `TransparentWrapper<T>` is not implemented for `Foo<T>`
    |         |
    |         required by a bound introduced by this call
    |
    = help: the trait `TransparentWrapper<T>` is implemented for `Wrapping<T>`
note: required by a bound in `constmuck::wrapper::peel_slice`
   --> src/wrapper.rs:278:12
    |
276 | pub const fn peel_slice<Outer, Inner>(reff: &[Outer]) -> &[Inner]
    |              ---------- required by a bound in this function
277 | where
278 |     Outer: TransparentWrapper<Inner>,
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `peel_slice`

error[E0277]: the trait bound `Foo<T>: TransparentWrapper<T>` is not satisfied
  --> tests/main_tests/ui/bound_wrapper_peel_fns_err.rs:33:55
   |
33 |         <Foo<T> as TransparentWrapper<T>>::peel_slice(b),
   |         --------------------------------------------- ^ the trait `TransparentWrapper<T>` is not implemented for `Foo<T>`
   |         |
   |         required by a bound introduced by this call
   |
   = help: the trait `TransparentWrapper<T>` is implemented for `Wrapping<T>`
